src/qmapshack/poi/CPoiDraw.cpp

```diff
@@ -137,6 +138,36 @@ void CPoiDraw::loadPoiPath(QSettings &cfg)
     poiPaths = cfg.value("poiPaths", poiPaths).toStringList();
 }
 
+poi_t CPoiDraw::findPOICloseBy(const QPoint &px) const
+{
+    poi_t poi;
+    CPoiItem::mutexActivePois.lock();
+    if(poiList)
+    {
+        for(int i = 0; i < poiList->count(); i++)
+        {
+            auto item = poiList->item(i);
+
+            if(!item || item->getPoifile().isNull())
+            {
+                // as all active maps have to be at the top of the list
+                // it is ok to break as soon as the first map with no
+                // active files is hit.
+                break;
+            }
+
+            item->getPoifile()->findPOICloseBy(px, poi);
```

Comment:
The only API exposed by item should be the one of `CPoiItem`. At this point I do not care if there is a poi file or what ever. I call `item->findPoiCloseBy()` and let the item do what ever it has to do.

Reply:
Should I somehow restrict access to `CPOIPoi::findPOICloseBy` or is it enough to just provide a `CPoiItem::findPOIClosBy`?

Reply:
Imho it's enough to provide a `CPoiItem::findPOIClosBy()`.